test/integration/eni-subnet-selection/eni_subnet_selection_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package eni_subnet_selection
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/resources/k8s/manifest"
+	"github.com/aws/amazon-vpc-cni-k8s/test/framework/utils"
+	"github.com/aws/amazon-vpc-cni-k8s/test/integration/common"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/apps/v1"
+)
+
+var _ = Describe("ENI Subnet Selection Test", func() {
+	var (
+		deployment      *v1.Deployment
+		podLabelKey     string
+		podLabelVal     string
+		err             error
+		newEniSubnetIds []string
+	)
+
+	Context("when creating deployment", func() {
+		BeforeEach(func() {
+			podLabelKey = "role"
+			podLabelVal = "eni-subnet-selection-test"
+		})
+
+		JustBeforeEach(func() {
+			By("creating deployment")
+			container := manifest.NewNetCatAlpineContainer(f.Options.TestImageRegistry).
+				Command([]string{"sleep"}).
+				Args([]string{"3600"}).
+				Build()
+
+			deploymentBuilder := manifest.NewBusyBoxDeploymentBuilder(f.Options.TestImageRegistry).
+				Container(container).
+				Replicas(50).
+				PodLabel(podLabelKey, podLabelVal).
+				NodeSelector("topology.kubernetes.io/zone", *primaryInstance.Placement.AvailabilityZone).
```

Comment:
shouldn't we target the node directly, as there could be multiple nodes in this availability zone